core/cache/application.go

```diff
@@ -28,18 +33,35 @@ type Application struct {
 	configHash string
 }
 
-func (m *Application) setDetails(details ApplicationChange) {
-	m.mu.Lock()
-	defer m.mu.Unlock()
-	m.details = details
+func (a *Application) WatchFields(comparitors ...func(*ApplicationDelta) bool) *ApplicationFieldWatcher {
+	return newApplicationFieldWatcher(a.hub, a.topic(applicationFieldChange), comparitors)
+}
+
+// TODO (manadart 2018-03-13) Should we change this up and down the call stack
+// so that pointers are passed instead of copies?
```

Comment:
most guidelines I've seen say copies are cheaper as long as the size of the object is small, as then you limit the pointer/gc/escape overhead (get more items on the stack instead of heap, ec). So if you don't need to modify the content, and the amount of content isn't particularly large, start with copies, and go to pointers only if you need it.

Reply:
I agree with this point. Especially given the heap fragmentation that it appears we have normally.